### PR TITLE
Bump github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring from 0.49.0 to 0.66.0

### PR DESCRIPTION
**Pull request by nak3**
This patch replaces with https://github.com/openshift-knative/serverless-operator/pull/2161, it needed to pin to sigs.k8s.io/controller-runtime version.

### CODE REVIEW

**Comment by ReToCode** [comment]
Rebased to have https://github.com/openshift-knative/serverless-operator/pull/2182 included.

File: go.mod [hunk 1]
```diff
@@ -23,24 +23,24 @@ require (
 	go.uber.org/zap v1.24.0
 	golang.org/x/sync v0.3.0
 	google.golang.org/grpc v1.56.1
-	k8s.io/api v0.25.4
-	k8s.io/apimachinery v0.25.4
-	k8s.io/client-go v0.25.4
+	k8s.io/api v0.27.2
```
Comment (ReToCode) [inline go.mod:26]: This should not be in here, right?

**Review by ReToCode — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (nak3) [reply go.mod:26]: It should be no problem. It is ignored as we pin to `k8s.io/api => k8s.io/api v0.25.4` - https://github.com/nak3/serverless-operator/blob/replace-2161/go.mod#L192

It is same with [`knative.dev/eventing v0.37.1`](https://github.com/nak3/serverless-operator/blob/replace-2161/go.mod#L29) on release-v1.10. It should be 0.36 on release-1.10 but just [replaces with the correct version](https://github.com/openshift-knative/serverless-operator/blob/release-1.30/go.mod#L29).

**Comment by nak3** [comment]
/cc @ReToCode @pierDipi @skonto 

Could you please take a look?

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**